### PR TITLE
173 improvement twitter authentification

### LINKED ISSUE #173: [IMPROVEMENT]: Twitter Authentification
### Description

Bind twitch authentication to the front-end and database.

### To do

Implementation of web & mobile authentication.
Implemented the registration of tokens in the database.
Verification of the functioning of twitter in AreaLoop.

### Priority

High

### Project parts

- [X] Server
- [X] Application
- [X] Web
- [ ] Automation

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by STom6**
# New feature
Bind twitch authentication to the front-end and database.
Implementation of web & mobile authentication.
Implemented the registration of tokens in the database.
# Updated feature
Verification of the functioning of twitter in AreaLoop.
# PICTURES:
Screenshot 2023-02-27 at 20 38 56
Screenshot 2023-02-27 at 20 39 33
Screenshot 2023-02-27 at 20 40 15

### CODE REVIEW

**Review by JohanCDev — DISMISSED** [review]
Ntr

File: Web/src/SettingsPage.js [hunk 1]
```diff
@@ -417,7 +417,37 @@ export default function SettingsPage(props) {
      * Empty for the moment
      * @function twitterConnexion
     */
-    function twitterConnexion() {
+    async function twitterConnexion() {
```
Comment (osvegn) [inline Web/src/SettingsPage.js:421]: missing async field in documentation

File: Web/readme.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
 ![React](https://img.shields.io/badge/react-%2320232a.svg?style=for-the-badge&logo=react&logoColor=%2361DAFB)
 ![JavaScript](https://img.shields.io/badge/javascript-%23323330.svg?style=for-the-badge&logo=javascript&logoColor=%23F7DF1E)
 
-# Web
+# Web Client
```
Comment (osvegn) [inline Web/readme.md]: Why this file has been updated ? There are some errors such as Detox framework and react-native mobile mentions

**Review by osvegn — CHANGES_REQUESTED** [review]
Some changes for the readme and the documentation, otherwise it seems good

**Review by CedricCORGE — DISMISSED** [review]
lgtm

File: Web/readme.md (hunk 2 above)
Comment (STom6) [reply Web/readme.md]: when Johan came to the branch and committed his readme file came too.

File: Web/src/SettingsPage.js (hunk 1 above)
Comment (STom6) [reply Web/src/SettingsPage.js:421]: done

**Review by JohanCDev — DISMISSED** [review]
lgtm

**Review by JohanCDev — APPROVED** [review]
ntr

**Review by CedricCORGE — APPROVED** [review]
lgtm

**Review by osvegn — APPROVED** [review]

**Pull request merged**